src/midend/atomics_op_lower.cpp

```diff
@@ -97,8 +97,27 @@ bool graphit::AtomicsOpLower::ApplyExprVisitor::lowerCompareAndSwap(mir::FuncExp
                         = mir::to<mir::TensorArrayReadExpr>(eq_expr->operands[0]);
                 std::string field_name = tensor_array_read_expr->getTargetNameStr();
                 mir::Type::Ptr field_type = mir_context_->getVectorItemType(field_name);
-                if (mir::isa<mir::ScalarType>(field_type)){
-                    mir::ScalarType::Ptr scalar_type = mir::to<mir::ScalarType>(field_type);
+
+                //TODO this is hack to access local array type since it is not registered in the context map
```

Comment:
This is fine for now we should figure out a way to move things away from the context anyway and what is the right way to do this.